src/plugin-host/plugin-host.ts

```diff
@@ -0,0 +1,79 @@
+import { AbstractPlugin } from "./plugin/plugin";
+import { PLUGIN_EVENT } from "./plugin-events/plugin-event-types";
+import { PrePostMessagePluginEventArguments } from "./plugin-events/event-arguments/pre-post-message-plugin-event-arguments";
+import { UserScoreChangedPluginEventArguments } from "./plugin-events/event-arguments/user-score-changed-plugin-event-arguments";
+import { LeaderboardResetPluginEventArguments } from "./plugin-events/event-arguments/leaderboard-reset-plugin-event-arguments";
+import { NoArgumentsPluginEventArguments } from "./plugin-events/event-arguments/no-arguments-plugin-event-arguments";
+import { ChatServices } from "./plugin-chat-services/chat-services";
+import { Chat } from "../chat/chat";
+import { plugins } from "../context-root";
+
+/**
+ * Class exposing the Plugin Host concept.
+ * Plugins are managed by a Plugin Host, which controls
+ * messaging to, from and between plugins. It also
+ * keeps track of the general state of a plugin.
+ */
+export class PluginHost
+{
+  /**
+   * Collection of plugins currently running.
+   */
+  public readonly Plugins: AbstractPlugin[];
+  private Services: () => ChatServices;
+
+  /**
+   * Create a new Plugin Host.
+   * The Plugin Host will not by itself try and
+   * find compatible plugins. Instead, it requests
+   * a list of plugins to be provided.
+   * @param _plugins List of plugins this PluginHost should manage.
+   */
+  constructor(_plugins: AbstractPlugin[]) 
+  {
+    this.Plugins = _plugins;
+
+    // Activate all plugins until a mechanism to enable/disable has been created.
```

Comment:
What is meant by this mechanism, exactly? The server admin setting plugins enabled or disabled via a config? But then what is the plugin names string in the config meant for? I thought that was meant for this very mechanism.

Reply:
This isn't relevant anymore. Plugins can now be enabled/disables by users.